frontend/src/settings/Preferences.jsx

```diff
@@ -1,16 +1,89 @@
 import * as React from 'react';
 import {
-  Text,
+  Alert, Button, Switch, Platform,
 } from 'react-native';
+import { LinearGradient } from 'expo-linear-gradient';
+import RNDateTimePicker from '@react-native-community/datetimepicker';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { useFocusEffect } from '@react-navigation/native';
 import {
-  BoilerplateView,
+  PreferencesView, CardTitle, PreferencesTitle,
 } from '../Themes';
 
 function Preferences() {
+  const [darkMode, setDarkMode] = React.useState(false);
+  const [Alarm, setAlarm] = React.useState(true);
+  const [twelveHour, setTwelveHour] = React.useState(true);
+  const [alarmTime, setAlarmTime] = React.useState(new Date());
+  const [show, setShow] = React.useState(false);
+
+  const onChange = (event, selectedDate) => {
+    const currentDate = selectedDate || alarmTime;
+    setShow(Platform.OS === 'ios');
+    setAlarmTime(currentDate);
+    Alert.alert(`Alarm set to: ${currentDate.toTimeString().split(' ')[0]}`);
+  };
+
+  React.useEffect(() => {
+    (async () => {
+      try {
+        setDarkMode(JSON.parse(await AsyncStorage.getItem('@darkMode')));
+        setAlarm(JSON.parse(await AsyncStorage.getItem('@Alarm')));
+        setTwelveHour(JSON.parse(await AsyncStorage.getItem('@twelveHour')));
+        setAlarmTime(new Date(JSON.parse(await AsyncStorage.getItem('@alarmTime'))));
+      } catch (e) {
+        console.log(e);
+      }
+    })();
+  }, []);
+
+  useFocusEffect(React.useCallback(() => (() => {
+    (async () => {
+      try {
+        await AsyncStorage.setItem('@darkMode', JSON.stringify(darkMode));
+        await AsyncStorage.setItem('@Alarm', JSON.stringify(Alarm));
+        await AsyncStorage.setItem('@twelveHour', JSON.stringify(twelveHour));
+        await AsyncStorage.setItem('@alarmTime', JSON.stringify(alarmTime));
```

Comment:
In cases like here, I'd recommend putting all of the states into one JSON object and storing just one object into AsyncStorage. As you can see here, every call to AsyncStorage has to be awaited. Compare awaiting for just one set or get of an AsyncStorage item, instead of having to set or get 4 items every single time. Especially when AsyncStorage is getting set 4 times every single time one of the states changes, even if the other 3 don't change. It would probably be best to only save these preferences when the component gets unmounted/blurred/unfocused. However, doing that is kinda annoying with React function components, but here's an example [here](https://github.com/williamchong256/immersive-sleep/blob/aaac9c06af2ccbc9834ef959f83369809381e50a/frontend/src/detailedData.jsx#L49) using Refs from lines 49 to 86. Granted you don't have to make any of these changes if you don't want to because when this gets integrated with AWS DataStore I'll implement this UI logic anyways.